### ISSUE TITLE
inaccurate codec default

### ISSUE THREAD

**Issue by mixmix** [flumedb/flumelog-offset#7]
README: 
```
  codec: {encode, decode} // defaults to a json codec
```

CODE:
```
function id (e) { return e }
...

  var codec = opts.codec || {encode: id, decode: id}
```

**Comment by mixmix** [comment]
I think the JSON encoding as defualt would be nice, because I just tried relying on that nice sounding default and it makes my initial flume-log-offset setup simple ...

**Comment by dominictarr** [comment]
this is technically a breaking change, but since this is a new module being pretty actively developed it's okay to expect chaos. probably will have to change _the tests_, but hmm, this is one less roadbump for new users (in a js context) optimizers can still do binary whatever if they want. WOULD MERGE.

**Comment by christianbundy** [comment]
Is the right solution to fix the readme or fix the code? I don't have a strong opinion, I just know that I care lots about the readme's accuracy.

**Comment by dominictarr** [comment]
I fixed the readme, because that was easier.

**Comment by dominictarr** [comment]
Also, I've since changed my mind about json. The default behavior for a flumelog is really something we want to be standard across all flumelogs, (so we can have drop-in compatibility) and buffer is the lowest common denomenator.

**Comment by dominictarr** [comment]
fixed in 3.4.4
